benches/bench.rs

```diff
@@ -0,0 +1,103 @@
+#![feature(test)]
+
+extern crate test;
+extern crate smoltcp;
+
+mod wire {
+    use test;
+    #[cfg(feature = "proto-ipv6")]
+    use smoltcp::wire::{Ipv6Address, Ipv6Repr, Ipv6Packet};
+    use smoltcp::wire::{Ipv4Address, Ipv4Repr, Ipv4Packet};
+    use smoltcp::phy::{ChecksumCapabilities};
+    use smoltcp::wire::{IpAddress, IpProtocol};
+    use smoltcp::wire::{TcpRepr, TcpPacket, TcpSeqNumber, TcpControl};
+    use smoltcp::wire::{UdpRepr, UdpPacket};
+
+    #[cfg(feature = "proto-ipv6")]
+    const SRC_ADDR: IpAddress = IpAddress::Ipv6(Ipv6Address([0xfe, 0x80, 0, 0, 0, 0, 0, 0,
+                                                             0, 0, 0, 0, 0, 0, 0, 1]));
+    #[cfg(feature = "proto-ipv6")]
+    const DST_ADDR: IpAddress = IpAddress::Ipv6(Ipv6Address([0xfe, 0x80, 0, 0, 0, 0, 0, 0,
+                                                             0, 0, 0, 0, 0, 0, 0, 2]));
+
+    #[cfg(not(feature = "proto-ipv6"))]
+    const SRC_ADDR: IpAddress = IpAddress::Ipv4(Ipv4Address([192, 168, 1, 1]));
+    #[cfg(not(feature = "proto-ipv6"))]
+    const DST_ADDR: IpAddress = IpAddress::Ipv4(Ipv4Address([192, 168, 1, 2]));
+
+    #[bench]
+    fn bench_emit_tcp(b: &mut test::Bencher) {
+
+        static PAYLOAD_BYTES: [u8; 400] =
+            [0x2a; 400];
+
+        b.iter(|| {
+            let repr = TcpRepr {
+                src_port:     48896,
+                dst_port:     80,
+                seq_number:   TcpSeqNumber(0x01234567),
+                ack_number:   None,
+                window_len:   0x0123,
+                control:      TcpControl::Syn,
+                max_seg_size: None,
+                payload:      &PAYLOAD_BYTES
+            };
+            let mut bytes = vec![0xa5; repr.buffer_len()];
```

Comment:
You're benchmarking the `Vec` allocator too...

Reply:
Good catch